### PR TITLE
merge oOffset and iOffset calculation into one block

### PR DESCRIPTION
**Pull request by tingxingdong**
resolves #___

Summary of proposed changes:
-  change the kernel generator
-  
-

### CODE REVIEW

File: library/src/device/generator/generator.kernel.hpp [hunk 1]
```diff
@@ -342,21 +347,25 @@ namespace StockhamGenerator
             /*=======================================================*/
             std::string loop;
             loop += "\tif(dim == 1){\n";
-            loop += "\t\t" + offset_name + " += counter_mod*" + stride_name + "[1];\n";
+            loop += "\t\t" + offset_name1 + " += counter_mod*" + stride_name1 + "[1];\n";
+            if (output == true) loop += "\t\t" + offset_name2 + " += counter_mod*" + stride_name2 + "[1];\n";
```
Comment (b-sumner) [inline library/src/device/generator/generator.kernel.hpp]: How will this work with the ";" at the end of the preceding line?

Comment (tingxingdong) [inline library/src/device/generator/generator.kernel.hpp]: This is the generator, it generates a line of code like this 
`
oOffset += counter_2*stride_out[3] + counter_1*stride_out[2] + counter_mod_1*stride_out[1]; `

Comment (b-sumner) [inline library/src/device/generator/generator.kernel.hpp]: I get that, but doesn't line 350 end the generated line with a ";".  And then if output is true, we will have a +... following the ";" in the generated code, which is legal, but doesn't actually do anything.

Comment (bragadeesh) [inline library/src/device/generator/generator.kernel.hpp]: @b-sumner I don't see the problem, there's the offset_name2 string before the '+=' , which would look like 'oOffset += ' in the generated line.
@tingxingdong you tested this code right?

Comment (tingxingdong) [inline library/src/device/generator/generator.kernel.hpp]: for the generated two lines of code. I do not see the "+" sign.   They should be correct

```
		iOffset += counter_mod*stride_in[1];
		oOffset += counter_mod*stride_out[1];
```

Comment (b-sumner) [inline library/src/device/generator/generator.kernel.hpp]: Sorry, you're right.  I misread the code.

**Pull request merged**